test-harness/src/test/groovy/com/netflix/counductor/integration/test/DoWhileSpec.groovy

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.counductor.integration.test
+
+import com.netflix.conductor.common.metadata.tasks.Task
+import com.netflix.conductor.common.metadata.tasks.TaskDef
+import com.netflix.conductor.common.run.Workflow
+import com.netflix.conductor.common.utils.TaskUtils
+import com.netflix.conductor.core.execution.WorkflowExecutor
+import com.netflix.conductor.dao.QueueDAO
+import com.netflix.conductor.service.ExecutionService
+import com.netflix.conductor.service.MetadataService
+import com.netflix.conductor.test.util.WorkflowTestUtil
+import com.netflix.conductor.tests.utils.TestModule
+import com.netflix.governator.guice.test.ModulesForTesting
+import spock.lang.Specification
+
+import javax.inject.Inject
+
+@ModulesForTesting([TestModule.class])
+class DoWhileSpec extends Specification {
+
+    @Inject
+    ExecutionService workflowExecutionService
+
+    @Inject
+    MetadataService metadataService
+
+    @Inject
+    WorkflowExecutor workflowExecutor
+
+    @Inject
+    WorkflowTestUtil workflowTestUtil
+
+    @Inject
+    QueueDAO queueDAO
+
+    def setup() {
+        workflowTestUtil.registerWorkflows("do_while_integration_test.json",
+                "do_while_multiple_integration_test.json")
+    }
+
+    def cleanup() {
+        workflowTestUtil.clearWorkflows()
+    }
+
+    def "Test workflow with a single iteration Do While task"() {
+        given: "Number of iterations of the loop is set to 1"
+        def workflowInput = new HashMap()
+        workflowInput['loop'] = 1
+
+        when: "The workflow is started"
+        def workflowInstanceId = workflowExecutor.startWorkflow("Do_While_Workflow", 1, "looptest", workflowInput, null, null)
+
+        and: "The workflow is retrieved"
+        def workflow = workflowExecutionService.getExecutionStatus(workflowInstanceId, true)
+
+        then: "Verify that workflow is running"
+        workflow
+        workflow.status == Workflow.WorkflowStatus.RUNNING
+        workflow.tasks.size() == 2
+
+        when: "Polling for first task"
+        def task = workflowExecutionService.getLastPollTask("integration_task_0", "integration.test.worker", null)
```

Comment:
nit: use intuitive variable names. Instead of using generic variable name `task`, something like `task0` or `integrationTask0` will be me more readable

Reply:
fixed